scipy/special/_basic.py

```diff
@@ -2791,6 +2791,12 @@ def perm(N, k, exact=False):
 
     """
     if exact:
+        if not (isscalar(N) and isscalar(k)):
+            raise ValueError("`N` and `k` must scalar integers be with `exact=True`.")
+        if not floor(N) == N and floor(k) == k:
```

Comment:
can we add some parentheses to make this more readable

Reply:
```suggestion
        if not (floor(N) == N and floor(k) == k):
```